pandas/core/style.py

```diff
@@ -0,0 +1,769 @@
+                cellstyle.append(
+                    {'props': props,
+                     'selector': "row%s_col%s" % (r, c)}
+                )
+            body.append(row_es)
+
+        # uuid required to isolate table styling from others
+        # in same notebook in ipnb
+        return dict(head=head, cellstyle=cellstyle, body=body, uuid=uuid,
+                    precision=self.precision, table_styles=table_styles,
+                    caption=caption)
+
+    def render(self, table_styles=None, caption=None, uuid=None,
+               precision=None, trim=True):
+        """
+        Render the built up styles to HTML
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        table_styles: list or None
+            defaults to ``self.table_styles``
+        caption: str or None
+            defaults to ``self.caption``
+        uuid: str or None
+            defaults to random uuid
+        precision: int or None
+            defaults to self.precision
+        trim: bool
+            Small optimization to not create classes for cells that
+            have no styles applied to them.
+
+        Returns
+        -------
+        rendered: str
+            the rendered HTML
+
+        Notes
+        -----
+        ``Styler`` objects have defined the ``_repr_html_`` method
+        which automatically calls ``self.render()`` when it's the
+        last item in a Notebook cell. When calling ``Styler.render()``
+        directly, wrap the resul in ``IPython.display.HTML`` to view
+        the rendered HTML in the notebook.
+        """
+        self._compute()
+        table_styles = table_styles or self.table_styles
+        caption = caption or self.caption
+        uuid = uuid or self.uuid
+        d = self._translate(table_styles=table_styles, caption=caption,
+                            uuid=uuid)
+        if trim:
+            # filter out empty styles, every cell will have a class
+            # but the list of props may just be [['', '']].
+            # so we have the neested anys below
+            trimmed = [x for x in d['cellstyle'] if
+                       any(any(y) for y in x['props'])]
+            d['cellstyle'] = trimmed
+        return self.template.render(**d)
+
+    def _update_ctx(self, attrs):
+        """
+        update the state of the Styler. Collects a mapping
+        of {index_label: ['<property>: <value>']}
+
+        attrs: Series or DataFrame
+        should contain strings of '<property>: <value>;<prop2>: <val2>'
+        Whitespace shouldn't matter and the final trailing ';' shouldn't
+        matter.
+        """
+        for row_label, v in attrs.iterrows():
+            for col_label, col in v.iteritems():
+                i = self.index.get_indexer([row_label])[0]
+                j = self.columns.get_indexer([col_label])[0]
+                for pair in col.rstrip(";").split(";"):
+                    self.ctx[(i, j)].append(pair)
+
+    def _copy(self, deepcopy=False):
```

Comment:
Should I remove this until there's a need for it? Not sure why I implemented it in the first place. And the current implementation is broken since I switched to the lazy style evaluation.